api/v1alpha1/validation.go

```diff
@@ -989,6 +1026,21 @@ func ensureAppName(app ApplicationProviderSpec, appType ApplicationProviderType)
 			return provider.Image, nil
 		}
 
+		return *app.Name, nil
+	case ArtifactApplicationProviderType:
+		provider, err := app.AsArtifactApplicationProviderSpec()
+		if err != nil {
+			return "", fmt.Errorf("invalid artifact application provider: %w", err)
+		}
+
+		// default name to artifact if not provided
+		if app.Name == nil {
```

Comment:
wondering if we should continue this pattern or just make this required for artifacts